go/vt/vttablet/tabletmanager/vreplication/last_error.go

```diff
@@ -48,15 +50,27 @@ func (le *lastError) record(err error) {
 	le.mu.Lock()
 	defer le.mu.Unlock()
 	if err == nil {
+		log.Infof("resetting last error")
 		le.err = nil
 		le.firstSeen = time.Time{}
+		le.lastSeen = time.Time{}
 		return
 	}
 	if !vterrors.Equals(err, le.err) {
+		log.Infof("got new last error %+v for %s, was %+v", err, le.name, le.err)
 		le.firstSeen = time.Now()
+		le.lastSeen = time.Now()
 		le.err = err
+	} else {
+		// same error seen
+		log.Infof("got same last error for %s seen %+v, last seen at %s after %dms", le.name, le.err, le.lastSeen, int(time.Since(le.lastSeen).Milliseconds()))
```

Comment:
I think this would be better (`%q` is especially nice when there's whitespace in the value so that it's clear what the variable value is):
```
log.Infof("Got the same last error for %q: %+v ; first seen at %s and last seen %dms ago", le.name, le.err, le.firstSeen, int(time.Since(le.lastSeen).Milliseconds()))
```